mojave/items/guns/energy.dm

```diff
@@ -14,105 +97,155 @@
 	desc = "A modified AER9 equipped with a refraction kit that spreads its bolts."
 	icon = 'mojave/icons/objects/guns/guns_world.dmi'
 	icon_state = "lasershotgun"
+	base_icon_state = "lasershotgun"
 	inhand_icon_state = "lasershotgun"
 	lefthand_file = 'mojave/icons/mob/inhands/weapons/guns_inhand_left.dmi'
 	righthand_file = 'mojave/icons/mob/inhands/weapons/guns_inhand_right.dmi'
 	ammo_type = list(/obj/item/ammo_casing/energy/ms13/laser/scatter)
 	cell_type = /obj/item/stock_parts/cell/ammo/mfc
 	w_class = WEIGHT_CLASS_HUGE
+	fire_delay = 6
 
 /obj/item/gun/energy/ms13/laser/plasma
 	name ="A3-20 Plasma Rifle"
 	icon = 'mojave/icons/objects/guns/guns_world.dmi'
 	inhand_icon_state = "plasmacarabine"
 	icon_state = "plasmacarabine"
+	base_icon_state = "plasmacarabine"
 	lefthand_file = 'mojave/icons/mob/inhands/weapons/guns_inhand_left.dmi'
 	righthand_file = 'mojave/icons/mob/inhands/weapons/guns_inhand_right.dmi'
 	desc = "A top of line miniaturized plasma caster built by REPCONN in the wake of the Z43-521P failure. It is supperior to all previous rifles to enter service in the USCC."
 	ammo_type = list(/obj/item/ammo_casing/energy/ms13/plasma)
-	cell_type = /obj/item/stock_parts/cell/ammo/mfc
+	cell_type = /obj/item/stock_parts/cell/ammo/pc
 	w_class = WEIGHT_CLASS_HUGE
+	fire_delay = 6
 
 /obj/item/gun/energy/ms13/laser/plasma/scatter
 	name = "A3e-20b Multiplas Rifle"
 	icon = 'mojave/icons/objects/guns/guns_world.dmi'
 	inhand_icon_state = "multiplas"
 	icon_state = "multiplas"
+	base_icon_state = "multiplas"
 	lefthand_file = 'mojave/icons/mob/inhands/weapons/guns_inhand_left.dmi'
 	righthand_file = 'mojave/icons/mob/inhands/weapons/guns_inhand_right.dmi'
 	desc = "A modified A3-20 plasma caster built by REPCONN equipped with a multicasting kit that creates multiple weaker clots."
 	ammo_type = list(/obj/item/ammo_casing/energy/ms13/plasma/scatter)
-	cell_type = /obj/item/stock_parts/cell/ammo/mfc
+	cell_type = /obj/item/stock_parts/cell/ammo/pc
+	fire_delay = 7
 
 /obj/item/gun/energy/ms13/laser/plasma/pistol
 	name ="MPL-A Plasma Pistol"
 	icon = 'mojave/icons/objects/guns/guns_world.dmi'
 	inhand_icon_state = "plasmapistol"
 	icon_state = "plasmapistol"
+	base_icon_state = "plasmapistol"
 	lefthand_file = 'mojave/icons/mob/inhands/weapons/guns_inhand_left.dmi'
 	righthand_file = 'mojave/icons/mob/inhands/weapons/guns_inhand_right.dmi'
 	desc = "A pistol-sized miniaturized plasma caster built by REPCONN. It fires heavy low penetration plasma clots."
 	ammo_type = list(/obj/item/ammo_casing/energy/ms13/plasma/pistol)
 	w_class = WEIGHT_CLASS_NORMAL
-	cell_type = /obj/item/stock_parts/cell/ammo/mfc
+	cell_type = /obj/item/stock_parts/cell/ammo/pc
+	fire_delay = 6
 
 //projectiles
 
 /obj/projectile/beam/ms13
 	speed = 0.2 //Vanilla tg is 0.8
-
+	var/damage_constant = 1
+/obj/projectile/beam/ms13/Range()
+	if(hitscan != TRUE)
+		return ..()
+	var/turf/location = get_turf(src)
+	if(!location)
+		return ..()
+	. = ..()
```

Comment:
Check out http://www.byond.com/docs/ref/#/proc/%2e

`.` is the return value from a proc - i.e you can have `return 1` or `. = 1` 
and `..()` is the parent proc

because all of the logic in this is just calling the parent proc by returning `..()` or setting `. = ..()`
we can just remove this entire range proc

Reply:
but then the previous range proc will take affect and be the same as the original one of this (fall of in the vacuum of space)

Reply:
nvm cant find one, I can omit it entirely